src/utils/injectFireReferrerHideableImage.js

```diff
@@ -28,23 +31,39 @@ const IFRAME_PROPS = {
   }
 };
 
-export default request => {
+export default ({
+  appendNode = appendNodeToDocument,
+  awaitSelector = awaitSelectorInDocument,
+  createNode = createNodeInDocument,
+  fireImage = fireImageInDocument,
+  removeNode = removeNodeInDocument
+} = {}) => {
+  const fireOnPage = fireImage;
+
+  let hiddenIframe;
+
   const createIframe = () => {
+    if (hiddenIframe) {
+      return Promise.resolve(hiddenIframe);
+    }
     return awaitSelector(BODY).then(([body]) => {
-      const iframe = createNode(IFRAME, IFRAME_ATTRS, IFRAME_PROPS);
-      return appendNode(body, iframe);
+      hiddenIframe = createNode(IFRAME, IFRAME_ATTRS, IFRAME_PROPS);
+      return appendNode(body, hiddenIframe);
     });
   };
 
   const fireInIframe = ({ src }) => {
     return createIframe().then(iframe => {
       const currentDocument = iframe.contentWindow.document;
-      return fireImage({ src, currentDocument }).then(() => {
+      return fireImage({ src, currentDocument }).catch(() => {
+        hiddenIframe = undefined;
         removeNode(iframe);
       });
```

Comment:
Why do we need to remove the iframe? There is a race condition here where the iframe is removed between another idSync's createIframe and fireImage, so that by the time fireImage is called, the iframe is no longer part of the document.